src/uu/du/src/du.rs

```diff
@@ -161,6 +164,44 @@ impl Stat {
             metadata,
         })
     }
+
+    /// Create a Stat using safe traversal methods with `DirFd` for the root directory
+    #[cfg(target_os = "linux")]
+    fn new_from_dirfd(
+        dir_fd: &DirFd,
+        full_path: &Path,
+        _options: &TraversalOptions,
+    ) -> std::io::Result<Self> {
+        // Get metadata for the directory itself using fstat
+        let safe_metadata = dir_fd.metadata()?;
+
+        // Create file info from the safe metadata
+        let file_info = safe_metadata.file_info();
```

Comment:
The comment is redundant.
```suggestion
        let file_info = safe_metadata.file_info();
```